src/tmuxp/workspace/builder.py

```diff
@@ -346,12 +347,22 @@ def iter_create_windows(self, session, append=False):
             except (KeyError, IndexError):
                 pass
 
+            if has_gte_version("3.0"):
+                environment = panes[0].get("environment", wconf.get("environment"))
+            else:
+                logging.warning(
```

Comment:
Wouldn't this show the warning for every window, even when the user hasn't set an environment?

Reply:
Maybe fine to remove the warning? (providing we move it to libtmux?)

re: https://github.com/tmux-python/tmuxp/pull/845/files#r1030592815

Reply:
You're right ... somehow one check went missing. Re-added them (and left the warning in place as explained in the other comment).